### PR TITLE
Add nested attributes processing

### PR DESCRIPTION
**Pull request by dbruns91**

### CODE REVIEW

File: src/test/java/org/odpi/openmetadata/adapters/connectors/integration/lineage/sample/sampleLineageEventProcessorTest.java [hunk 1]
```diff
@@ -27,71 +27,70 @@
 
 
 /**
- *  Test of the event parsing into the EventContent object
+ * Test of the event parsing into the EventContent object
  */
-public class sampleLineageEventProcessorTest
-{
+public class SampleLineageEventProcessorTest {
```
Comment (davidradl) [inline src/test/java/org/odpi/openmetadata/adapters/connectors/integration/lineage/sample/sampleLineageEventProcessorTest.java]: this rename of the class - causes it to mismatch the file name so it does not build

**Comment by davidradl** [comment]
@dbruns91 the dependancy error is that you need to change the gradle file to 
` api "com.fasterxml.jackson.core:jackson-databind"` from implementation. You must be using that library in the test code

**Comment by dbruns91** [comment]
@davidradl Thanks! I updated it.

**Comment by davidradl** [comment]
@dbruns91 I see https://github.com/odpi/egeria-connector-integration-lineage-event-driven-sample/issues/67 with and without your fix, when I test against version 4. Are you testing against Egeria version4 or 3?

I you are looking to have this work against version4 - you need to make the change listed in the release guid https://egeria-project.org/release-notes/4-0/#release-40-march-2023 to do with the required package name change for v4?

**Comment by dbruns91** [comment]
@davidradl we are running on 3.15 ... I can update the sample-data connector but this will not work in our open shift environment at the moment. @T4ylan Is there any progress on updating egeria to v4?

**Comment by davidradl** [comment]
@dbruns91 I am setup to test on v4, there seems to be a regression as per the issue #67. So I am not in a position to easily test as sample1 fails for me. Does any of the test files work for you with the fix on?

**Comment by dbruns91** [comment]
@davidradl I am answering in #67 so we do not mix things up.

**Comment by davidradl** [comment]
@dbruns91 note any server side Egeria fix is likely only going to be done at the latest Egeria v4 level. Are you in a position to move to using Egeria v4?

**Comment by dbruns91** [comment]
@davidradl As a mentioned yesterday egeria base (and all our connectors) run on 3.15 at the moment. We need some time to update all of them.

**Comment by davidradl** [comment]
> @davidradl As a mentioned yesterday egeria base (and all our connectors) run on 3.15 at the moment. We need some time to update all of them.

@dbruns91 
OK. So your plan is:
- debug this sample connector at 3.15. I think we need a really well defined diagnosis of the issue. some questions:
   - Do any of the samples work after your restructure of the code? 
   - Does anything get created in Egeria for sample1? I assume the assets must be created. 
   - What fails with sample1 and what is the symptom with the latest code. 
   - is the issue that you are now using the parent call for all cases and this fails?  
 
- if we think a server side change is required then I assume you need to migrate to v4. Unless you want to fix Egeria core at a back level and release that.

**Comment by dbruns91** [comment]
@davidradl I tried the Sample1.json (even the input makes not really sense) and it looks good with my code changes:
```
Tue Jun 13 11:34:47 GMT 2023 cocoMDS1 Information LINEAGE_SAMPLE-INTEGRATION-CONNECTOR-0006 Event to be processed {"Id":"1234567890","Name":"TestRes","Description":"TestResource","Bounded Context":"TestResource","Team":"ITG","Input":[{"id":"C6B7B1B717C840F686EE2426241ED18CE1D053019534F03495E8CD644976FA37","name":"Foo","type":"SQL","value":"select * from foo;"}],"Output":[{"id":"vertriebskunde-services.agree-kundendaten","name":"Kundendaten","schema-type":"json-schema","schemas":[{"$id":"https://example.com/person.schema.json","$schema":"https://json-schema.org/draft/2020-12/schema","id":"Person","type":"object","properties":{"firstName":{"type":"string","description":"The person's first name."},"lastName":{"type":"string","description":"The person's last name."},"age":{"description":"Age in years which must be equal to or greater than zero.","type":"integer"},"derivedAge":{"description":"Derived Age.","type":"integer","formula":"test-formula"},"derivedAge2":{"description":"Derived Age.","type":"integer","...
Tue Jun 13 11:35:25 GMT 2023 cocoMDS1 Information LINEAGE_SAMPLE-INTEGRATION-CONNECTOR-0007 Event successfully processed
```
When I try the Sample1.json again I get the following error:
```
Tue Jun 13 11:47:06 GMT 2023 cocoMDS1 Exception LINEAGE_SAMPLE-INTEGRATION-CONNECTOR-0003 Exception org.odpi.openmetadata.frameworks.connectors.ffdc.PropertyServerException occurred in Integration connector LineageIntegrator - the error message was OMAG-GENERIC-HANDLERS-500-001 An unsupported bean class named org.odpi.openmetadata.accessservices.assetmanager.metadataelements.SchemaTypeElement was passed to the repository services by the getSchemaTypeForElement request for open metadata access service Asset Manager OMAS on server mds1; error message was: class org.odpi.openmetadata.accessservices.assetmanager.metadataelements.SchemaTypeElement cannot be cast to class org.odpi.openmetadata.accessservices.assetmanager.properties.SchemaTypeProperties (org.odpi.openmetadata.accessservices.assetmanager.metadataelements.SchemaTypeElement and org.odpi.openmetadata.accessservices.assetmanager.properties.SchemaTypeProperties are in unnamed module of loader org.springframework.boot.loader.LaunchedURLClassLoader @46da...
```
--> https://github.com/odpi/egeria/issues/7692

I do not need to debug the sample connector - already happened, mutliple times. The bug is in egeria base. At frist I want to try to change the audit log (of egeria base) to get more information, then debug through it ... but you are right a migration to v4 is necessary @T4ylan @juergenhemelt

**Comment by davidradl** [comment]
@dbruns91 ok I think you are saying that if you run the sample1 once then issue call "`{{mds-baseURL}}/servers/{{mds-server}}/open-metadata/access-services/asset-manager/users/{{user}}/parents/{{asset-type}}/{{asset-guid}}/schema-types/retrieve?forLineage=true&forDuplicateProcessing=true`" specifying the assed GUID it will fail.

**Comment by dbruns91** [comment]
@mandy-chessell can you review this? 
I think there is one problem with the gradle buildHealth:
```
Advice for root project
Unused dependencies which should be removed:
  api 'org.odpi.egeria:open-integration-framework:4.2-SNAPSHOT'
```
When I remove this I have compilation errors.

**Comment by mandy-chessell** [comment]
Try using the compileOnly option

**Pull request merged**